rpcs3/Crypto/unpkg.cpp

```diff
@@ -295,6 +295,8 @@ bool pkg_install(const std::string& path, atomic_t<double>& sync)
 		decrypt(0, header.file_count * sizeof(PKGEntry), header.pkg_platform == PKG_PLATFORM_TYPE_PSP ? PKG_AES_KEY2 : dec_key.data());
 	}
 
+	bool install_success = true;
```

Comment:
Can initialize this to false instead and assign to true in line 378 currently.
Also a corner case to think about: what if all entries name size are above 256, no entry would be installed yet you are returning success.

Reply:
That will not work, as setting it to `true` at line 378 would set it so after every successfully installed file, even if there were errors earlier. However, starting with `true` and setting it to `false` in line 310 will cover the case you're describing.

Reply:
I corrected myself.

Reply:
`install_success` is checked after all files are processed, so this way it would **still** fail only if the last file failed. Or do I misunderstand?

Reply:
Can also add a counter which increments on entry installation success, then check if it matches entries.size() after the loop.

Reply:
Sure, but does it matter right now? I am positive all error cases are now covered by the flag.

That said, it would be potentially beneficial to count successes as then I could say "X file(s) failed to install" :)

Reply:
Imo successes based check is better than rasing a flag on each failure as it prevents misusing/forgetting using this flag after potential rewrites/refactoring in the future.

Reply:
I agree. Do you think it makes sense to tell the user how many files failed to install? IMO it's a bit redundant and there's no way user could make a good use of this info, as everything is already listed in the log.

Reply:
Nah I think its TMI.

Reply:
Solved. I am counting failures after all since they are more obvious - they are always accompanied `LOG_ERROR`, meanwhile the opposite is less obvious - installation can still succeed when some files are not overwritten (`did_overwrite && (entry.type & PKG_FILE_ENTRY_OVERWRITE) == 0`).

I also added `bool extract_success` to remove a false positive "Created file X" line from the log in case extracting or writing failed.